[CostModel] Remove some negative costs.

davemgreen opened the pull request:
The cost model in the past returned -1 for unknown costs, but over time this has largely been removed. This cleans up some of the uses that have remained. It uses 0/free for the cost of an insert and 1/basic for the cost of anything that is unknown.

RKSimon reviewed (commented):
Why not use InstructionCost::getInvalid() ?

davemgreen commented:
> Why not use InstructionCost::getInvalid() ?

An Insert I would not expect to be invalid, and for the one at the end of getInstructionCost, if it can return a valid cost for !Recip returning the same cost as a fallback sounded OK to me. An Invalid cost to me means "this cannot be codegenerated" (or maybe just "strongly don't do this" at times), not "I don't know".

lukel97 reviewed (approved):
LGTM. Looks like those RISC-V insertvalues get lowered to COPYs which may or may not end up as vmv.v.vs, or otherwise a register rename.

RKSimon commented on llvm/include/llvm/Analysis/TargetTransformInfoImpl.h:
```diff
@@ -755,8 +755,9 @@ class TargetTransformInfoImplBase {
     // Note: The `insertvalue` cost here is chosen to match the default case of
     // getInstructionCost() -- as pior to adding this helper `insertvalue` was
```
pior -> prior

RKSimon reviewed (approved):
OK - LGTM (with a trivial minor)

davemgreen committed:
[CostModel] Remove some negative costs.

The cost model in the past returned -1 for unknown costs, but over time this
has largely been removed. This cleans up some of the uses that have remained.
It uses 0/free for the cost of an insert and 1/basic for the cost of anything
that is unknown.
